### ISSUE TITLE
Arc - Method used by Yasson not implemented

### ISSUE THREAD

**Issue by masini** [quarkusio/quarkus#1656]
Adding a @JsonbTypeSerializer or other annotations that lookup for a Bean, this Exception is thrown:

```
Caused by: java.lang.UnsupportedOperationException
	at io.quarkus.arc.BeanManagerImpl.createAnnotatedType(BeanManagerImpl.java:211)
	at org.eclipse.yasson.internal.components.BeanManagerInstanceCreator.lambda$getOrCreateComponent$0(BeanManagerInstanceCreator.java:68)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at org.eclipse.yasson.internal.components.BeanManagerInstanceCreator.getOrCreateComponent(BeanManagerInstanceCreator.java:67)
	at org.eclipse.yasson.internal.ComponentMatcher.introspectSerializerBinding(ComponentMatcher.java:292)
	at org.eclipse.yasson.internal.AnnotationIntrospector.getSerializerBinding(AnnotationIntrospector.java:299)
	at org.eclipse.yasson.internal.AnnotationIntrospector.introspectCustomization(AnnotationIntrospector.java:694)
	at org.eclipse.yasson.internal.MappingContext$ParseClassModelFunction.apply(MappingContext.java:54)
	at org.eclipse.yasson.internal.MappingContext$ParseClassModelFunction.apply(MappingContext.java:37)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at org.eclipse.yasson.internal.MappingContext.getOrCreateClassModel(MappingContext.java:107)
	at org.eclipse.yasson.internal.Marshaller.marshall(Marshaller.java:73)
	at org.eclipse.yasson.internal.JsonBinding.toJson(JsonBinding.java:98)
	at org.jboss.resteasy.plugins.providers.jsonb.JsonBindingProvider.writeTo(JsonBindingProvider.java:133)
	... 60 more
```

Is documented that createAnnotatedType is not supported but JSONB RI uses it as well (I think is older than Quarkus itself).

**Comment by gsmet** [comment]
@mkouba any chance we could implement that method?

**Comment by mkouba** [comment]
We don't plan to fully support `Annotated` abstraction layer, the only exception is injection point metadata for `@Dependent` beans. Also `InjectionTarget` API is not supported ATM.  Ideally, Yasson should provide some SPI so that we can use ArC API. In this particular case, it should be quite straightforward.

**Comment by gsmet** [comment]
@mkouba Well, they do use the CDI SPI for that. Not sure they will want to include a separate one.

What's the plan for this, then? The use case is fairly common IMHO.

**Comment by mkouba** [comment]
> Not sure they will want to include a separate one.

Well, it doesn't have to be a separate one but a more general one so that it would be possible to use any DI framework.

> What's the plan for this, then? The use case is fairly common IMHO.

We don't have a plan yet. So far our strategy was not to support all "traditional" APIs (for example portable extensions) but provide an alternative way to do the same stuff.

**Comment by mkouba** [comment]
For the record - I've created https://github.com/eclipse-ee4j/yasson/issues/260.

**Comment by Edubits** [comment]
FYI: Yasson 1.0.4 is available in Maven central containing the fix for eclipse-ee4j/yasson#260.
